src/GuiBase/Viewer/Gizmo/GizmoManager.cpp

```diff
@@ -1,22 +1,50 @@
+#include <Engine/Managers/SystemDisplay/SystemDisplay.hpp>
+#include <Engine/Renderer/Camera/Camera.hpp>
 #include <Engine/Renderer/OpenGL/OpenGL.hpp>
+#include <GuiBase/Utils/KeyMappingManager.hpp>
+#include <GuiBase/Utils/Keyboard.hpp>
 #include <GuiBase/Viewer/CameraInterface.hpp>
 #include <GuiBase/Viewer/Gizmo/GizmoManager.hpp>
 #include <GuiBase/Viewer/Gizmo/RotateGizmo.hpp>
 #include <GuiBase/Viewer/Gizmo/ScaleGizmo.hpp>
 #include <GuiBase/Viewer/Gizmo/TranslateGizmo.hpp>
 
-#include <Engine/Managers/SystemDisplay/SystemDisplay.hpp>
-#include <Engine/Renderer/Camera/Camera.hpp>
-
-#include <GuiBase/Utils/KeyMappingManager.hpp>
-
 namespace Ra {
 namespace Gui {
+
+///\todo If a macro expert could write a recursive marco to have something like
+/// KM_DEFINE_STATICS(GizmoManager, KeyMappingGizmo)
+#define KMA_VALUE( XX ) Gui::KeyMappingManager::KeyMappingAction GizmoManager::XX;
+KeyMappingGizmo;
+#undef KMA_VALUE
+
+void GizmoManager::registerKeyMapping() {
+    m_keyMappingContext = Gui::KeyMappingManager::getInstance()->getContext( "GizmoContext" );
+    if ( m_keyMappingContext.isInvalid() )
+    {
+        LOG( Ra::Core::Utils::logINFO )
+            << "GizmoContext not defined (maybe the configuration file do not contains it";
+    }
+#define KMA_VALUE( XX ) \
+    XX = Gui::KeyMappingManager::getInstance()->getActionIndex( m_keyMappingContext, #XX );
+    KeyMappingGizmo
+#undef KMA_VALUE
```

Comment:
In case the context is invalid we should return and not try to access it, otherwise this will crash